### PR TITLE
DOC: Fixed redirects in various parts of the documentation

### PR DESCRIPTION
**Pull request by lrjball**
Links in the docs to part way down a page were being sent to the top of the page due to a generic redirect from pandas.pydata.org/pandas-doc/stable/... -> pandas.pydata.org/pandas-doc/stable/user_guide/... . These links have been change to go straight to the .../user_guide/... version of the page to ensure the link goes to the correct part of the docs. 

For example, the link within the `chunksize` parameter of pandas.read_json() currently goes to the top of the IO Tools page, when the relevant section is the 'Line delimited json' heading about half way down.

### CODE REVIEW

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -272,7 +272,8 @@ class Categorical(ExtensionArray, PandasObject):
     Notes
     -----
     See the `user guide
-    <http://pandas.pydata.org/pandas-docs/stable/categorical.html>`_ for more.
+    <http://pandas.pydata.org/pandas-docs/stable/user_guide/categorical.html>`_
```
Comment (TomAugspurger) [inline pandas/core/arrays/categorical.py]: Can we use refs for these instead instead? I think this would be, e.g. :ref:\`categorical\`.

**Review by TomAugspurger — COMMENTED** [review]
Can you change the absolute URLs in the docstrings to refs?

The ones in error / warning messages will need to continue to be URLs.

File: pandas/core/arrays/categorical.py (hunk 1 above)
Comment (topper-123) [reply pandas/core/arrays/categorical.py]: I like the explicit urls in the doc strings: If in doubt after reading a doc string, I can just copy/paste the url into a browser....couldn't do that with a ref.

Comment (lrjball) [reply pandas/core/arrays/categorical.py]: It looks like both are used in various parts of the documentation, although :ref: does seem a bit more common now that I look. Is there an agreed best practice?

**Comment by TomAugspurger** [comment]
I prefer the :ref: syntax, as it will be updated if the section title
changes. Though it does have the the downside mentioned
by @topper-123 when viewing the raw docstring.

**Comment by lrjball** [comment]
This pull request is to fix the links which were going to the wrong place. It seems like whether :ref: or urls are used is a separate issue, as urls are used all over the documentation, not just in the places I have fixed here. Maybe replacing urls with :ref: could be raised as it's own issue.

**Review by WillAyd — APPROVED** [review]
I'm OK with this as is and follow up PR to convert to ref.

@TomAugspurger

**Comment by TomAugspurger** [comment]
Sure.

**Comment by WillAyd** [comment]
Thanks @lrjball !

**Comment by lrjball** [comment]
Cheers
 @TomAugspurger I'm happy to pick up a PR to change all links to refs unless you have already picked it up.

**Comment by TomAugspurger** [comment]
I don't think anyone is working on that.

**Pull request merged**